File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/repository/ParkingRepository.java
package com.lattis.ellipse.domain.repository;

import com.lattis.ellipse.data.network.model.response.parking.GetParkingFeeForFleetResponse;
import com.lattis.ellipse.domain.model.Location;
import com.lattis.ellipse.domain.model.Parking;

import java.util.List;

import io.reactivex.Observable;


public interface ParkingRepository {

    Observable<List<Parking>> findParkings(double latitude, double longitude, float rangeInMiles);

    Observable<List<Parking>> getParkingSpotsForFleet(int fleetId);

    Observable<GetParkingFeeForFleetResponse> getParkingFeeForFleet(Location location,
                                                                    int fleet_id);

}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/repository/ParkingRepository.kt
package com.lattis.domain.repository

import com.lattis.domain.models.*
import io.reactivex.rxjava3.core.Observable

interface ParkingRepository {
    fun getParkingFeeForFleet(
        location: Location,
        fleet_id: Int
    ): Observable<ParkingFeeForFleet>

    fun getParkingSpotsForFleet(fleetId: Int): Observable<List<Parking>>
    fun getParkingZone(fleetID: Int): Observable<List<ParkingZone>>
    fun getDockHub(bikeId:Int,location: Location):Observable<List<DockHub>>
    fun getGeoFences(fleetID: Int): Observable<List<GeoFence>>
}